### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by FallenBY1**

### CODE REVIEW

File: task1/src/GenreToggle.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+export const GenreToggle = () => {
+    return (
```
Comment (DrDiman) [inline task1/src/GenreToggle.js:4]: can be without return.

U can leave as is)

File: task1/src/SearchForm.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+export class SearchForm extends React.PureComponent {
+    constructor() {
+        super();
+        this.state = {
```
Comment (DrDiman) [inline task1/src/SearchForm.js:6]:
```
state = {
  keyword: '',
  showResults: false
}
```

File: task2/.babelrc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (DrDiman) [inline task2/.babelrc]: u will commit to this `task2` folder then?

File: task2/index.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>React App</title>
+</head>
+<body>
+<noscript>You need to enable JavaScript to run this app.</noscript>
+<div id="root"></div>
```
Comment (DrDiman) [inline task2/index.html]: something with indents..

File: task2/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react'
```
Comment (DrDiman) [inline task2/src/index.tsx]: wrote somewhere.. that it can be omitted.

File: task2/webpack.config.dev.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      {
+        test: /\.s[ac]ss$/i,
+        use: [stylesHandler, "css-loader", "sass-loader"],
+      },
+      {
+        test: /\.(eot|svg|ttf|woff|woff2|png|jpg|gif)$/i,
+        type: "asset",
+      },
+      {
+        test: /\.(gif|png|jpe?g|svg)$/i,
```
Comment (DrDiman) [inline task2/webpack.config.dev.js]: why did u use two loaders for images?  as `asset` and with `file-loader` and `image-webpack-loader`?

File: task2/webpack.config.dev.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
+// Generated using webpack-cli https://github.com/webpack/webpack-cli
+
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const stylesHandler = "style-loader";
```
Comment (DrDiman) [inline task2/webpack.config.dev.js]: why did u decide to create this const?) looks like as overkill

File: task2/webpack.config.prod.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    module: {
+        rules: [
+            {
+                test: /\.html$/,
+                exclude: /node_modules/,
+                use: [
+                    {
+                        loader: 'html-loader',
+                        options: {
+                            sources: false,
```
Comment (DrDiman) [inline task2/webpack.config.prod.js]: use plz same approach with indents everywhere

File: task2/webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+module.exports = process.env.NODE_ENV === 'production'
```
Comment (DrDiman) [inline task2/webpack.config.js]: take a look at 'webpack-merge' npm module

good practice to have 3 files for webpack:

prod, dev, common 

webpack.common.js will contain common configs for both prod and dev - e.g. loaders

File: task2/webpack.config.prod.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+        open: true
+    },
+    optimization: {
+        splitChunks: {
+            chunks: 'all',
+        }
+    },
+    plugins: [
+        new CleanWebpackPlugin(),
+        new ESLintPlugin(),
```
Comment (DrDiman) [inline task2/webpack.config.prod.js]: what is it for here? 

let's talk

File: task2/src/app.tsx [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react'
```
Comment (DrDiman) [inline task2/src/app.tsx]: ?

File: task2/webpack.config.dev.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        test: /\.css$/i,
+        use: [stylesHandler, "css-loader"],
+      },
+      {
+        test: /\.s[ac]ss$/i,
+        use: [stylesHandler, "css-loader", "sass-loader"],
+      },
+      {
+        test: /\.(eot|svg|ttf|woff|woff2|png|jpg|gif)$/i,
+        type: "asset",
```
Comment (DrDiman) [inline task2/webpack.config.dev.js]: "asset/resource"

File: task1/src/GenreToggle.js (hunk 1 above)
Comment (FallenBY1) [reply task1/src/GenreToggle.js:4]: fixed

**Pull request merged**